Core/Dialog/PSPNetconfDialog.cpp

```diff
@@ -374,11 +456,30 @@ int PSPNetconfDialog::Shutdown(bool force) {
 void PSPNetconfDialog::DoState(PointerWrap &p) {	
 	PSPDialog::DoState(p);
 
-	auto s = p.Section("PSPNetconfigDialog", 0, 1);
+	auto s = p.Section("PSPNetconfigDialog", 0, 2);
 	if (!s)
 		return;
 
 	Do(p, request);
+	if (s >= 2) {
+		Do(p, scanInfosAddr);
+		Do(p, scanStep);
+		Do(p, connResult);
+	}
+	else {
+		scanInfosAddr = 0;
+		scanStep = 0;
+		connResult = -1;
+	}
+
+	if (p.mode == p.MODE_READ) {
+		// Reallocate if it wasn't the correct memory block
```

Comment:
You definitely shouldn't need to do this, unless `scanInfosAddr` is 0.  The memory blocks are all loaded as well, although you shouldn't depend on the sequence of loading.

For example, `PSPNetconfDialog::DoState` may actually be called before `userMemory.DoState` (I think this is not the case, but I don't think the code should depend on the order.)  You really should not interact with `userMemory` within this function at all.

Once all DoState functions finish, you can depend on `userMemory` looking exactly like it did when the user saved state.  So the address will still be valid if it was valid before.

When a save state is "upgraded" then you need a fallback, but that's it.  And in that case, `scanStep` will also be 0 so we should be safe.

-[Unknown]

Reply:
Btw, does kernelMemory blocks also saved just like userMemory? i remembered long time ago i use kernelMemory for the loop code instead of userMemory, may be i should change that to userMemory too